digits/extensions/data/imageSegmentation/forms.py

```diff
@@ -0,0 +1,121 @@
+            pass
+        else:
+            # make sure the filesystem path exists
+            if not os.path.exists(field.data) or not os.path.isdir(field.data):
+                raise validators.ValidationError(
+                    'Folder does not exist or is not reachable')
+            else:
+                return True
+
+    def validate_file_path(form, field):
+        if not field.data:
+            pass
+        else:
+            # make sure the filesystem path exists
+            if not os.path.exists(field.data) and not os.path.isdir(field.data):
+                raise validators.ValidationError(
+                    'File does not exist or is not reachable')
+            else:
+                return True
+
+    feature_folder = utils.forms.StringField(
+        u'Feature image folder',
+        validators=[
+            validators.DataRequired(),
+            validate_folder_path,
+            ],
+        tooltip="Indicate a folder full of images."
+        )
+
+    label_folder = utils.forms.StringField(
+        u'Label image folder',
+        validators=[
+            validators.DataRequired(),
+            validate_folder_path,
+            ],
+        tooltip="Indicate a folder full of images. For each image in the feature"
+                " image folder there must be one corresponding image in the label"
+                " image folder. The label image must have the same filename except"
+                " for the extension, which may differ. Label images are expected"
+                " to be single-channel images (paletted or grayscales)."
+        )
+
+    folder_pct_val = utils.forms.IntegerField(
+        u'% for validation',
+        default=10,
+        validators=[
+            validators.NumberRange(min=0, max=100)
+            ],
+        tooltip="You can choose to set apart a certain percentage of images "
+                "from the training images for the validation set."
+        )
+
+    has_val_folder = utils.forms.BooleanField('Separate validation images',
+            default = False,
+            )
+
+    validation_feature_folder = utils.forms.StringField(
+        u'Validation feature image folder',
+        validators=[
+            validate_required_iff(has_val_folder=True),
+            validate_folder_path,
+            ],
+        tooltip="Indicate a folder full of images."
+        )
+
+    validation_label_folder = utils.forms.StringField(
+        u'Validation label image folder',
+        validators=[
+            validate_required_iff(has_val_folder=True),
+            validate_folder_path,
+            ],
+        tooltip="Indicate a folder full of images. For each image in the feature"
+                " image folder there must be one corresponding image in the label"
+                " image folder. The label image must have the same filename except"
+                " for the extension, which may differ. Label images are expected"
+                " to be single-channel, palette images."
```

Comment:
To match the above desc, should this be "single-channel images (paletted or grayscale)"?